core/actionProxy/actionproxy.py

```diff
@@ -286,6 +300,18 @@ def complete(response):
 
 
 def main():
+    platformImpl = None
+    PlatformFactory.addPlatform(PLATFORM_OPENWHISK, OpenWhiskImpl)
+    PlatformFactory.addPlatform(PLATFORM_KNATIVE, KnativeImpl)
```

Comment:
Love the use of the factory pattern, but realistically we are built for either one platform or the other.  For perf. sake we should not "addPlatform" for platforms other than the on set in the environment on load. Shaving milliseconds...

Down  the road if we support other target platforms or if the proxy can change platform impls. dynamically (on the fly), the factory reg. approach will be appreciated.

Reply:
I have taken direct use of the Platform factory but left it in as a comment for future committers.

Reply:
@pwplusnick Sounds like a good strategy in case we decide to add support for other platforms and allow change of platform emulation between requests (perhaps useful for reuse/concurrency).